holoviews/core/spaces.py

```diff
@@ -759,6 +759,27 @@ def _cache(self, key, val):
         self.data[key] = val
 
 
+    def map(self, map_fn, specs=None, clone=True):
+        """
+        Recursively replaces elements using a map function when the
+        specification applies.
+        """
+        if specs and not isinstance(specs, list): specs = [specs]
+        applies = specs is None or any(self.matches(spec) for spec in specs)
+
+        if applies:
+            deep_mapped = map_fn(self)
+        else:
+            deep_mapped = self.clone() if clone else self
+        for k, v in deep_mapped.data.items():
+            deep_mapped[k] = v.map(map_fn, specs, clone)
```

Comment:
It seems in the non-dynamic version we first map the items and then apply it to itself if it applies, so I'll do that here.